client/src/components/layout/ReviewTile.js

```diff
@@ -1,12 +1,57 @@
-import React from "react";
+import React, { useState, useRef, useEffect } from "react";
 
 const ReviewTile = (props) => {
+  const inputRef = useRef(null);
+  const [inputVisible, setInputVisible] = useState(false);
+  const [comments, setComments] = useState(props.review.comments);
+
+  const deleteReviewHandler = () => {
+    props.deleteReview(props.review);
+  };
+
+  function onClickOutSide(e) {
+    if (inputRef.current && !inputRef.current.contains(e.target)) {
+      setInputVisible(false);
+    }
+  }
+
+  useEffect(() => {
+    if (inputVisible) {
+      document.addEventListener("mousedown", onClickOutSide);
+    }
+
+    return () => {
+      document.removeEventListener("mousedown", onClickOutSide);
+    };
```

Comment:
This direct DOM manipulation is not ideal in React, we want our components to be controlled components by React through state and user interaction